samanthas_telegram_bot/conversation/callbacks/teacher.py

```diff
@@ -0,0 +1,214 @@
+
+
+async def store_readiness_to_host_speaking_clubs_ask_additional_help_or_bye(
+    update: Update, context: CUSTOM_CONTEXT_TYPES
+) -> int:
+    """If teacher is ready to host speaking clubs, asks for additional skills, else says bye.
+
+    This callback is for young teachers only.
+    """
+    query, data = await answer_callback_query_and_get_data(update)
+    locale = context.user_data.locale
+
+    await query.delete_message()
+
+    if data == CommonCallbackData.YES:  # yes, I can host speaking clubs
+        context.user_data.teacher_can_host_speaking_club = True
+        await update.effective_chat.send_message(
+            PHRASES["ask_teacher_any_additional_help"][locale]
+        )
+        return ConversationState.ASK_FINAL_COMMENT
+
+    await update.effective_chat.send_message(PHRASES["reply_cannot_work"][locale])
+    return ConversationHandler.END
+
+
+async def store_communication_language_ask_teaching_experience(
+    update: Update, context: CUSTOM_CONTEXT_TYPES
+) -> int:
+    """Callback for teachers. Stores communication language, asks about teaching experience."""
+
+    (
+        query,
+        context.user_data.communication_language_in_class,
+    ) = await answer_callback_query_and_get_data(update)
+
+    if context.chat_data["mode"] == ConversationMode.REVIEW:
+        await query.delete_message()
+        await MessageSender.ask_review(update, context)
+        return ConversationState.REVIEW_MENU_OR_ASK_FINAL_COMMENT
+
+    logger.info(context.user_data.communication_language_in_class)
+
+    await CQReplySender.ask_yes_no(
+        context, query, question_phrase_internal_id="ask_teacher_experience"
+    )
+    return ConversationState.ASK_TEACHING_GROUP_OR_SPEAKING_CLUB
+
+
+async def store_experience_ask_about_groups_or_speaking_clubs(
+    update: Update, context: CUSTOM_CONTEXT_TYPES
+) -> int:
+    """Stores if teacher has experience, asks teaching preferences (groups vs speaking clubs)."""
+
+    query, data = await answer_callback_query_and_get_data(update)
+
+    context.user_data.teacher_has_prior_experience = (
+        True if data == CommonCallbackData.YES else False
+    )
+
+    await CQReplySender.ask_teacher_can_teach_regular_groups_speaking_clubs(context, query)
+    return ConversationState.ASK_NUMBER_OF_GROUPS_OR_TEACHING_FREQUENCY_OR_NON_TEACHING_HELP
+
+
+async def store_teaching_preference_ask_groups_or_frequency_or_student_age(
+    update: Update, context: CUSTOM_CONTEXT_TYPES
+) -> int:
+    """Stores information about teaching preferences, next action depends on several factors:
+
+    * If teacher only wants to host speaking clubs, asks about preferred student age groups
+    * If teacher can teach regular groups but has no experience, asks about frequency
+    * If teacher can teach regular groups and has experience, asks about number of groups
+    """
+
+    query, data = await answer_callback_query_and_get_data(update)
+
+    context.user_data.teacher_can_host_speaking_club = (
```

Comment:
`context.user_data.teacher_can_host_speaking_club = data in (TeachingMode.SPEAKING_CLUB_ONLY, TeachingMode.BOTH)`

Reply:
Done